### PR TITLE
Adds types for react-native-side-menu

### PR DESCRIPTION
**Pull request by matt-pawley**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present). 

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
/cc @jsamr (you're named in the DT header, apologies to bug you)
Thanks!

File: types/react-native-side-menu/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Type definitions for react-native-side-menu 1.1
+// Project: react-native-side-menu
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:2]:
```suggestion
// Project: https://github.com/react-native-community/react-native-side-menu#readme
```
I'd suggest this, to help someone decide on update,etc (source-of-the-truth)

**Comment by jsamr** [comment]
Great initiative. My PR https://github.com/react-native-community/react-native-side-menu/pull/376 is over 1 year old! I'm closing it then.

File: types/react-native-side-menu/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { GestureResponderEvent, PanResponderGestureState, Animated, ViewStyle } from 'react-native';
+
+export interface ReactNativeSideMenuProps {
+    /**
+     * Menu component
+     */
+    menu: ReactNode;
+    /**
+     * Props driven control over menu open state
+     */
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:17]:
```suggestion
     * @default false
     */
```

File: types/react-native-side-menu/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     * X axis tolerance
+     */
+    toleranceX?: number;
+    /**
+     * Y axis tolerance
+     */
+    toleranceY?: number;
+    /**
+     * Disable whether the menu can be opened with gestures or not
+     */
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:41]:
```suggestion
     * @default false
     */
```

File: types/react-native-side-menu/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    disableGestures?: boolean;
+    /**
+     * Function that accepts event as an argument and specify if side-menu should react on the touch or not.
+     * Check https://facebook.github.io/react-native/docs/gesture-responder-system.html for more details
+     */
+    onStartShouldSetResponderCapture?: (e: GestureResponderEvent, gestureState: PanResponderGestureState) => boolean;
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:47]: Not my area, the docs suggests single event passed, is that correct?

File: types/react-native-side-menu/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    onChange?: (isOpen: boolean) => void;
+    /**
+     * Callback on menu move. Is passed left as an argument
+     */
+    onMove?: (left: number) => void;
+    /**
+     * Callback when menu is sliding. It returns a decimal from 0 to 1 which represents the percentage of menu offset between hiddenMenuOffset and openMenuOffset.
+     */
+    onSliding?: (fraction: number) => void;
+    menuPosition?: 'left' | 'right';
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:60]:
```suggestion
     /**
     * @default 'left'
     */
    menuPosition?: 'left' | 'right';
```

File: types/react-native-side-menu/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    animationFunction?: (prop: Animated.Value, value: number) => Animated.CompositeAnimation;
+    animationStyle?: (value: number) => ViewStyle;
+    /**
+     * When true, content view will bounce back to openMenuOffset when dragged further
+     */
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:65]:
```suggestion
     * @default true
     */
```

File: types/react-native-side-menu/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    bounceBackOnOverdraw?: boolean;
+    /**
+     * When true, menu close automatically as soon as an event occurs
+     */
```
Comment (peterblazejewicz) [inline types/react-native-side-menu/index.d.ts:69]:
```suggestion
     * @default true
     */
```

**Review by peterblazejewicz — APPROVED** [review]
Seems `onAnimationComplete` callback is missing (if we are on the same page with source, btw):
https://github.com/react-native-community/react-native-side-menu/search?q=onAnimationComplete&unscoped_q=onAnimationComplete

Otherwise, with minor changes, all good!
LGTM!
@toughdeveloper thx for contribution!

**Comment by toughdeveloper** [comment]
@peterblazejewicz - Thanks for feedback + your original PR in the library repo. 

I've added the changes you suggested if you don't mind taking another look

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@toughdeveloper thx for the contribution!

**Pull request merged**